gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,384 @@
+      String inputName,
+      Map<String, String> properties,
+      String projectId,
+      String region,
+      BigLakeClient client) {
+    this.name = inputName;
+    this.bigLakeProperties = ImmutableMap.copyOf(properties);
+    this.bigLakeProjectId = projectId;
+    this.bigLakeRegion = region;
+    Preconditions.checkNotNull(client, "BigLake client must not be null");
+    this.bigLakeClient = client;
+
+    // Users can specify the BigLake catalog ID, otherwise catalog plugin name will be used.
+    // For example, "spark.sql.catalog.<plugin-name>=org.apache.iceberg.spark.SparkCatalog"
+    // specifies the plugin name "<plugin-name>".
+    this.catalogId =
+        this.bigLakeProperties.getOrDefault(GCPProperties.BIGLAKE_CATALOG_ID, inputName);
+    this.catalogName = CatalogName.of(projectId, region, catalogId);
+
+    String ioImpl =
+        this.bigLakeProperties.getOrDefault(
+            CatalogProperties.FILE_IO_IMPL, ResolvingFileIO.class.getName());
+    this.io = CatalogUtil.loadFileIO(ioImpl, this.bigLakeProperties, conf);
+
+    this.closeableGroup = new CloseableGroup();
+    closeableGroup.addCloseable(io);
+    closeableGroup.setSuppressCloseFailure(true);
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier identifier) {
+    return new BigLakeTableOperations(
+        bigLakeClient,
+        io,
+        name(),
+        tableName(databaseId(identifier.namespace()), identifier.name()));
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier identifier) {
+    String locationUri = loadDatabase(identifier.namespace()).getHiveOptions().getLocationUri();
+    return String.format(
+        "%s/%s",
+        Strings.isNullOrEmpty(locationUri)
+            ? databaseLocation(databaseId(identifier.namespace()))
+            : locationUri,
+        identifier.name());
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    ImmutableList<DatabaseName> dbNames;
+    if (namespace.isEmpty()) {
+      dbNames =
+          Streams.stream(bigLakeClient.listDatabases(catalogName))
+              .map(db -> DatabaseName.parse(db.getName()))
+              .collect(ImmutableList.toImmutableList());
+    } else {
+      dbNames = ImmutableList.of(databaseName(namespace));
+    }
+
+    ImmutableList.Builder<TableIdentifier> result = ImmutableList.builder();
+    dbNames.stream()
+        .map(
+            dbName ->
+                Streams.stream(bigLakeClient.listTables(dbName))
+                    .map(BigLakeCatalog::tableIdentifier)
+                    .collect(ImmutableList.toImmutableList()))
+        .forEach(result::addAll);
+    return result.build();
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    TableOperations ops = newTableOps(identifier);
+    // TODO: to catch NotFoundException as in https://github.com/apache/iceberg/pull/5510.
+    TableMetadata lastMetadata = ops.current();
```

Comment:
I think this should init `ops` and read current tablemetadata only if `purge` is set to `true`

Reply:
Done.